File: ui/src/commands/npc/precursor.ts
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

import { invoke } from '@tauri-apps/api/core';

export function getPublicPrecursor(id: PrecursorId) {
  return invoke<PublicPrecursor>('get_public_precursor', { id });
}

File: crates/nil/src/command/npc/precursor.rs
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

use crate::error::Result;
use crate::manager::ManagerExt;
use nil_core::npc::precursor::{PrecursorId, PublicPrecursor};
use tauri::AppHandle;

#[tauri::command]
pub async fn get_public_precursor(app: AppHandle, id: PrecursorId) -> Result<PublicPrecursor> {
  app
    .client(async |cl| cl.get_public_precursor(id).await)
    .await?
    .map_err(Into::into)
}
